storage/shard.go

```diff
@@ -673,16 +744,13 @@ func (s *dbShard) insertSeriesEntries(inserts []dbShardInsert) error {
 		// Insert still pending, perform the insert
 		entry = inserts[i].entry
 		s.lookup[entry.series.ID().Hash()] = s.list.PushBack(entry)
-
-		// If we are going to write to this entry then increment the
-		// writer count so it does not look empty immediately after
-		// we release the write lock
-		if inserts[i].hasPendingWrite {
-			entry.incrementWriterCount()
-		}
 	}
 	s.Unlock()
 
+	if !anyPendingWrites {
+		return nil
+	}
+
```

Comment:
Async inserts by their very nature do not really have a good place to surface.  I can add a metric however.